src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -701,6 +815,104 @@ public virtual int ReadByte()
             return r == 0 ? -1 : oneByteArray[0];
         }
 
+        /// <summary>
+        /// Reads bytes from the current stream and advances the position within the stream until the <paramref name="buffer"/> is filled.
+        /// </summary>
+        /// <param name="buffer">A region of memory. When this method returns, the contents of this region are replaced by the bytes read from the current stream.</param>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="buffer"/> was an empty buffer.
+        /// </exception>
+        /// <exception cref="EndOfStreamException">
+        /// Thrown when the end of the stream is reached before filling the <paramref name="buffer"/>.
+        /// </exception>
+        public void ReadExactly(Span<byte> buffer)
+        {
+            if (buffer.IsEmpty)
+            {
+                ThrowHelper.ThrowArgumentException_BufferMustNotBeEmpty(nameof(buffer));
+            }
+
+            _ = ReadAtLeastCore(buffer, buffer.Length, throwOnEndOfStream: true);
+        }
+
+        /// <summary>
+        /// Reads <paramref name="count"/> number of bytes from the current stream and advances the position within the stream.
+        /// </summary>
+        /// <param name="buffer">An array of bytes. When this method returns, the buffer contains the specified byte array with the values between offset and (offset + count - 1) replaced by the bytes read from the current stream.</param>
```

Comment:
Usually I don't mind about the length of the comment lines, but to keep it consistent with the rest of your comments, I suggest we split this.

```suggestion
        /// <param name="buffer">An array of bytes. When this method returns, the buffer contains the specified byte array with the values
        /// between <paramref name="offset"/> and (<paramref name="offset"/> + <paramref name="count"/> - 1) replaced by the bytes read from the current stream.</param>
```